src/query/api/v1/handler/prometheus/native/read_common.go

```diff
@@ -45,6 +48,15 @@ func read(
 	ctx, cancel := context.WithTimeout(reqCtx, params.Timeout)
```

Comment:
Maybe delete this ctx if we're not gonna use it? 

`_, cancel := context.WithTimeout(reqCtx, params.Timeout)`